agi-bin/directory.lib.php

```diff
@@ -189,9 +189,9 @@ function readContact($con, $keys='#'){
 				//fallthough if not successfull
 			case 'tts':
 				// speak the name if possible, otherwise move on to spell it
-				$temporaryAudioFile = $this->agi_get_var('ASTSPOOLDIR') . '/tmp/directory-tts-' . time() . rand(100, 999) . 'wav';
-				system('flite -t "' . escapeshellarg($con['name']) .'" -o ' . $temporaryAudioFile, $exitCode);
-				if (file_exists($temporaryAudioFile) && $exitCode === 0) {
+				$temporaryAudioFile = $this->agi_get_var('ASTSPOOLDIR') . '/tmp/directory-tts-' . time() . rand(100, 999);
+				system('flite -t "' . escapeshellarg($con['name']) .'" -o ' . $temporaryAudioFile . '.wav', $exitCode);
+				if (file_exists($temporaryAudioFile . '.wav') && $exitCode === 0) {
 					$ret = $this->agi->stream_file($temporaryAudioFile, $keys);
 					$ret['result'] = isset($ret['result']) ? chr($ret['result']) : NULL;
 					unlink($temporaryAudioFile);
```

Comment:
Thanks wizhippo, can't believe I missed the unlink :(
The break 2 is something that was already in the file, so I left it in.
I'll remove the pull request so your patch is merged.